drivers/gles3/storage/texture_storage.cpp

```diff
@@ -1196,11 +1197,17 @@ void TextureStorage::_update_render_target(RenderTarget *rt) {
 		rt->fbo = system_fbo;
 		return;
 	}
-
-	rt->color_internal_format = rt->is_transparent ? GL_RGBA8 : GL_RGB10_A2;
-	rt->color_format = GL_RGBA;
-	rt->color_type = rt->is_transparent ? GL_BYTE : GL_UNSIGNED_INT_2_10_10_10_REV;
-	rt->image_format = Image::FORMAT_RGBA8;
+	if (rt->force_high_precision && config->framebuffer_float_supported) {
+		rt->color_internal_format = rt->is_transparent ? GL_RGBA16F : GL_RGB16F;
+		rt->color_format = GL_RGBA;
+		rt->color_type = GL_BYTE;
```

Comment:
Type should be ``GL_HALF_FLOAT`` and I am now just realizing I missed something else. We need to check for half float support as well not just float support as the [GL_EXT_color_buffer_float](https://registry.khronos.org/OpenGL/extensions/EXT/EXT_color_buffer_float.txt) extension only makes RGBA16F a renderable format, it does not include RGB16F, for that you need [GL_EXT_color_buffer_half_float](https://registry.khronos.org/OpenGL/extensions/EXT/EXT_color_buffer_half_float.txt). So ``framebuffer_float_supported`` should indicate that both GL_EXT_color_buffer_float and GL_EXT_color_buffer_half_float are available.

Reply:
Made the fix. I didn't knew that `RGB16F` format was considered so exotic. I could try to find a way to avoid it in this case but i'm not sure how yet.
Thanks again for the carefull feedbacks !